cdk/checkfiles_runner/stacks/runner.py

```diff
@@ -0,0 +1,267 @@
+            PolicyStatement(
+                actions=[
+                    'ssm:SendCommand',
+                    'ssm:GetCommandInvocation',
+                ],
+                resources=['*'],
+            )
+        )
+
+        run_checkfiles_command_lambda.add_to_role_policy(
+            PolicyStatement(
+                actions=[
+                    'logs:CreateLogGroup',
+                    'logs:CreateLogStream',
+                    'logs:PutLogEvents',
+                    'logs:DescribeLogStreams',
+                ],
+                resources=['*'],
+            )
+        )
+
+        run_checkfiles_command = LambdaInvoke(
+            self,
+            'RunCheckFilesCommand',
+            lambda_function=run_checkfiles_command_lambda,
+            payload_response_only=True,
+            result_selector={
+                'instance_id.$': '$.instance_id',
+                'command_id.$': '$.command_id'
+            }
+        )
+
+        wait_for_checkfiles_lambda = PythonFunction(
+            self,
+            'WaitForCheckfilesLambda',
+            entry='checkfiles_runner/lambdas/wait_checkfiles',
+            runtime=Runtime.PYTHON_3_11,
+            index='main.py',
+            handler='wait_checkfiles_command_to_finish',
+            timeout=Duration.seconds(180),
+        )
+
+        wait_for_checkfiles_lambda.add_to_role_policy(
+            PolicyStatement(
+                actions=[
+                    'ssm:GetCommandInvocation'
+                ],
+                resources=['*'],
+            )
+        )
+
+        wait_for_checkfiles = LambdaInvoke(
+            self,
+            'WaitForCheckfiles',
+            lambda_function=wait_for_checkfiles_lambda,
+            payload_response_only=True,
+            result_selector={
+                'checkfiles_command_status.$': '$.checkfiles_command_status',
+                'instance_id.$': '$.instance_id',
+                'command_id.$': '$.command_id',
+                'instance_id_list.$': '$.instance_id_list'
+            }
+        )
+
+        wait_for_checkfiles.add_retry(
+            backoff_rate=1,
+            errors=['CommandInProgress'],
+            interval=Duration.seconds(3600),
+            max_attempts=12,
+        )
+
+        no_files_to_process = Succeed(
+            self,
+            'No files to process.'
+        )
+
+        terminate_instance = CallAwsService(
+            self,
+            'TerminateInstance',
+            service='ec2',
+            action='terminateInstances',
+            iam_resources=['*'],
+            parameters={
+                'InstanceIds.$': '$.instance_id_list'
+            },
+            result_path=JsonPath.DISCARD,
+        )
+
+        definition = check_pending_files.next(
+            Choice(self, 'Pending files?').when(
+                Condition.boolean_equals('$', False), no_files_to_process
+            ).otherwise(
+                create_checkfiles_instance.next(
```

Comment:
Does this keep creating new checkfile instances every time it's run if there are still pending files? Even if the old checkfiles instance is still running?

Reply:
Yes, it will create and terminate one instance per state machine run. In reality the old instance should never be up when the new execution starts (For example on a 16cpu machine we should be able to handle ~1000 files in 24h, and we are not going to get even close to that many submissions in any 24h period).

Reply:
Never say never. Worse case what happens if more than one instance is running at a time?

Reply:
The second instance would skip patching files that the first instance patches when the second one is in progress of checking. If something gets checked/patched multiple times that is fine, operation is idempotent.